EIPS/eip-6260.md

```diff
@@ -0,0 +1,290 @@
+| \_tokenId | uint representing the token ID number.                  |
+| \_price   | uint representing the price at which to sell the token. |
+
+#### **removeTokenSale**
+
+Removes a token from the sale.
+
+| Parameter | Description                            |
+| --------- | -------------------------------------- |
+| \_tokenId | uint representing the token ID number. |
+
+#### **buyToken**
+
+Buys a specific token from its ID onchain.
+
+| Parameter | Description                            |
+| --------- | -------------------------------------- |
+| \_tokenId | uint representing the token ID number. |
+
+#### **\_royaltyDenominator**
+
+The denominator to interpret the rate of royalties, defaults to 10000 so rate are expressed in basis points. _May be customized with an override._
+
+#### **\_defaultRoyalty**
+
+Royalty percentage per default at the contract creation expressed in basis points (between 0 and `_royaltyDenominator` and 1000 per default).
+
+#### **\_royalty**
+
+Return the current royalty without its denominator.
+
+#### **royaltyInfo**
+
+Return the current royalty and its denominator.
+
+#### **setRoyalty**
+
+Set the royalty percentage.
+
+| Parameter    | Description                                                             |
+| ------------ | ----------------------------------------------------------------------- |
+| \_newRoyalty | uint within the range of `_royaltyDenominator` as new tokens royalties. |
+
+#### **\_payRoyalties**
+
+Send to `_owner` of the contract a specific amount of ether as royalties.
+
+| Parameter | Description                   |
+| --------- | ----------------------------- |
+| \_amount  | uint for the royalty payment. |
+
+## Rationale
+
+- Fully on-chain transaction condition
+
+We wanted to make it possible for a token to be sold and buyable within its own contract so that their holders don't have to interact with any other third parties and take the risk of being affected by a vulnerability within the selling condition as it was recently seen with Opensea that had their marketplace conduct off-chain listing in order to save gas. A middleman (whether on-chain or off-chain) sometimes adds unnecessary extra steps + approvals when the blockchain actually allows direct peer-to-peer exchanges with total trust thanks to the conditions written in the smart-contract.
+
+- Royalties matter
+
+Regarding the royalties payment, NFT creators rely on marketplaces like OpenSea & Rarible to collect it when their creations are sold. Over time, we discussed the idea of making this entire process completely decentralised. With such solution, artists would be allowed to 100% enforce their due royalties without taking the risk of trusting a marketplace to pay them directly nor having to specify to each marketplace one by one the % they want to collect while being constrained to be capped. Our solution by being able to directly sell a token onchain makes it easy to do just that.
```

Comment:
These paragraphs are more suited to the Motivation section. The Rationale section should explain individual technical choices made within the proposal itself, while the Motivation section justifies the entire EIP as a whole.